### PR TITLE
Code Cleanup

### LINKED ISSUE #43: Code cleanup
- [x] Maybe rework whitelist file read/write with own wrapper struct
- [x] Move more things to the config
- [x] Make separate Data directory (Maybe seperate Data struct to read/write from/to the data directory
- [x] Splitting up files for better organisation
- [x] Make more values in the config optional
- [x] Make helper functions etc. generic
- [x] Remove unused sqlx dependency
- [x] Fix "unused return value of `std::option::Option::<T>::insert` that must be used" in discord.rs
- [x] Implement tests for ClassesAndUsers struct
- [x] Fix #47 
- [x] (Last) Update Dependencies

### LINKED ISSUE #47: Possible panic on register
There is a possible panic when registering for a class.
Occurs when the virtual cursor in the class whitelist json file is not at the beginning when deserializing.
Gets fixed by #46

### PR DESCRIPTION
**Pull request by flamion**

### CODE REVIEW

File: src/discord_notifier.rs [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+		let _ = second.block_1.insert("ONE".to_owned());
+		let _ = second.block_2.insert("TWO".to_owned());
+		let _ = second.block_3.insert("THREE".to_owned());
+		let _ = second.block_4.insert("FOUR".to_owned());
+		let _ = second.block_5.insert("FIVE".to_owned());
+		table_map.insert("SECOND".to_owned(), &second);
+
+		let out = DiscordNotifier::table_from_substitutions(&table_map);
+
+		let expected_1 = "\
```
Comment (FlareFlo) [inline src/discord_notifier.rs:204]: Maybe use include_string!() here instead of filling src

**Review by FlareFlo — APPROVED** [review]
Very naice!

File: src/substitution_schedule.rs [hunk 2]
```diff
@@ -175,7 +177,7 @@ impl SubstitutionSchedule {
 			.collect();
 
 		#[allow(clippy::cast_possible_wrap)]
-		let date = chrono::Date::<Local>::from_utc(
+			let date = chrono::Date::<Local>::from_utc(
```
Comment (zUnixorn) [inline src/substitution_schedule.rs:180]: i liked the former indentation more

File: src/discord_notifier.rs [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+
+use crate::classes_and_users::ClassesAndUsers;
+use crate::commands::{after, before, dispatch_error, Handler, normal_message, unknown_command};
+use crate::commands::*;
+use crate::config::Config;
+use crate::SOURCE_URLS;
+use crate::substitution_pdf_getter::Weekdays;
+use crate::substitution_schedule::{Substitutions, SubstitutionSchedule};
+
+pub trait Notifier {
```
Comment (zUnixorn) [inline src/discord_notifier.rs:24]: Consider implementing or removing it

File: src/data.rs [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+use crate::substitution_pdf_getter::Weekdays;
+use crate::TypeMapKey;
+
+const PDF_JSON_DIR_NAME: &str = "pdf_jsons";
+const WHITELIST_JSON_FILE_NAME: &str = "class_whitelist.json";
+const CLASSES_AND_USERS_FILE_NAME: &str = "class_registry.json";
+
+pub struct Data {
+	data_directory: String,
+	whitelist_file: Mutex<File>,
```
Comment (zUnixorn) [inline src/data.rs:17]: i like i like

**Review by zUnixorn — APPROVED** [review]

**Pull request merged**